### PR TITLE
[WebProfilerBundle] fix compatibility with Twig 3.10

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | 
| License       | MIT

see twigphp/Twig#4055

### CODE REVIEW

File: src/Symfony/Bundle/WebProfilerBundle/Tests/Twig/WebProfilerExtensionTest.php [hunk 1]
```diff
@@ -23,7 +24,7 @@ class WebProfilerExtensionTest extends TestCase
      */
     public function testDumpHeaderIsDisplayed(string $message, array $context, bool $dump1HasHeader, bool $dump2HasHeader)
     {
-        $twigEnvironment = $this->mockTwigEnvironment();
```
Comment (stof) [inline src/Symfony/Bundle/WebProfilerBundle/Tests/Twig/WebProfilerExtensionTest.php:26]: the unused private method `mockTwigEnvironment` should be removed

File: src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php [hunk 2]
```diff
@@ -114,10 +115,15 @@ public function getName()
 
     private static function escape(Environment $env, string $s): string
     {
+        // Twig 3.9
         if (method_exists(EscaperExtension::class, 'escape')) {
             return EscaperExtension::escape($env, $s);
         }
 
+        if (class_exists(EscaperRuntime::class) && null !== $escaperExtension = $env->getRuntime(EscaperRuntime::class)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php]: Let's move this above the previous check and add a comment about the version

Comment (xabbuh) [reply src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php]: done

File: src/Symfony/Bundle/WebProfilerBundle/Tests/Twig/WebProfilerExtensionTest.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Bundle/WebProfilerBundle/Tests/Twig/WebProfilerExtensionTest.php:26]: oh indeed, good catch

File: src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php [hunk 3]
```diff
@@ -114,6 +115,12 @@ public function getName()
 
     private static function escape(Environment $env, string $s): string
     {
+        // Twig 3.10 and above
+        if (class_exists(EscaperRuntime::class) && null !== $escaperRuntime = $env->getRuntime(EscaperRuntime::class)) {
```
Comment (derrabus) [inline src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php]: Do we actually need the `class_exists()` check?

Comment (xabbuh) [reply src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php]: Without we get this error:

> Twig\Error\RuntimeError: Unable to load the "Twig\Runtime\EscaperRuntime" runtime.

Comment (derrabus) [reply src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php]: Oh, and `getRuntime()` does not return `null` if the requested runtime cannot be loaded: it raises an exception instead.

Comment (derrabus) [reply src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php]: Right, so the `null` check is unnecessary then.

Comment (xabbuh) [reply src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php]: Ah indeed, I was confused by the behaviour of the mock we used before.

**Review by derrabus — APPROVED** [review]

**Review by alexandre-daubois — APPROVED** [review]

**Comment by derrabus** [comment]
Thank you Christian.

File: src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php (hunk 3 above)
Comment (stof) [reply src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php]: this is what allows detecting Twig 3.10+

**Pull request merged**